tests/regression/33-constants/03-empty-not-dead-branch.c

```diff
@@ -0,0 +1,16 @@
+//PARAM: --set ana.activated '["constants"]'
+
+int g;
+
+int main() {
+    // state: {bot}, because no locals
+    if (g) {
+        g = 1; // state: {bot}, because Hoare set add (in PathSensitive2 map) keeps bot, while reduce would remove
+    }
+    else {
+        g = 2; // state: {bot}, because Hoare set add (in PathSensitive2 map) keeps bot, while reduce would remove
+    }
+    // state: {}, because reduce is applied after join and that removes bot
+    g = 3; // state: {} and PathSensitive2 map raises Deadcode
```

Comment:
Can we enhance this test in some way so we would notice if this became bot again through some changes?

Reply:
Also, the comment seems outdated?

Reply:
I'll fix the comments because, yes, they correspond to the previous problematic state.

Not sure how we could enhance this test, because there's no way to test reachability here. Normally when base is active, `assert(1)` works for that purpose but constants analysis doesn't do anything about asserts. And the same test doesn't work with base, because it would have `g` in the local state, so it wouldn't be bottom.

An option that I've thought about is extracting an assert output analysis from base into a standalone analysis which just queries others (just like base does right now and after #251). Then I guess constants can run just with assert analysis and respond to `EvalInt` queries with the constant value (which constants can evaluate).